src/modules/hmisAppSettings/Provider.tsx

```diff
@@ -23,7 +25,22 @@ export const HmisAppSettingsProvider: React.FC<Props> = ({ children }) => {
     if (fetched?.appName) document.title = fetched.appName;
   }, [fetched]);
 
-  if (error) throw error;
+  useEffect(() => {
+    if (error) {
+      const cause = error?.cause as any;
+      if (cause?.error?.type === 'timeout') {
+        // Refetch settings if the session timed out
+        fetchHmisAppSettings().then(setFetched).catch(setError);
```

Comment:
Why do we ned to fetch it again? If the session is already timed out, wont we just get the same error?

Reply:
No actually. It will fetch fine the second time. My guess is that the session gets nulled on the frontend after it gets the timeout response and then the next request will have no session which will pass correctly.

Also we have to fetch it again because the app settings aren't returned with the error response so we do need to go and try to fetch them again.

Reply:
Huh, odd.